### DISCUSSION TITLE
vercel nextjs app with Dynamodb access denied on production.

### DISCUSSION THREAD

**Discussion by yasen002** [vercel/community#452] [Help] [answered]
Hi, 
I have a Nextjs app deployed on Vercel. When I run it locally, it doesn't have any error, but when I deploy it, it has access denied error. 

This is how I set up my ddb_client
`import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
// Set the AWS Region.
const REGION = "REGION"; //e.g. "us-east-1"
// Create an Amazon DynamoDB service client object.
const ddbClient = new DynamoDBClient({
    accessKeyId: process.env.ACCESS_KEY,
    secretAccessKey: process.env.SECRET_KEY,
    region: process.env.REGION,
    // endpoint: "http://localhost:8000",
});
export { ddbClient };
`

Here is the error on my vercel function real time logs:
`I have .env.local file with env variables on my localhost and added env variables through vercel dashboard.

error: AccessDeniedException: User: arn:aws:sts::00359*******:assumed-role/cloudwatch_logs_events_putter/..... is not authorized to perform: dynamodb:PutItem on resource: arn:aws:dynamodb:us-east-1:.........:table/table_name
   ......................................
 {
    __type: 'com.amazon.coral.service#AccessDeniedException',
    '$fault': 'client',
    '$metadata': {
      httpStatusCode: 400,`

I used my admin IAM account to create a user with [AmazonDynamoDBFullAccess] permission.

In the page, I used js Fetch to call nextjs API:
`const submitResponse = await fetch(`${baseUrl}/api/form/submit`, {
                method: 'POST',
                mode: 'cors',
                cache: 'no-cache',
                credentials: 'same-origin',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({ formData })`

I am not getting any errors locally, why am I getting errors on my vercel deployed app, has anyone had this error before?

**Comment by jedwards4044** [comment]
I am running into the same thing. Did you ever resolve this issue?

**Reply by yasen002** [reply]
I wasn't able to solve this. I think I ended up using Amplify or the newer version of AWS SDK.

**Reply by leerob** [reply, marked as answer]
I have an example here you can check out! https://github.com/leerob/nextjs-aws-dynamodb

**Reply by asfand987** [reply]
@jedwards4044 did you figure it out?

**Comment by asfand987** [comment]
All of a sudden my NextJS has started getting this, it was working completely fine for months before. Did not make any changes. Works fine on my local...

**Reply by amyegan** [reply]
Hey @asfand987. It could be an expired environment variable. That's the first thing I would check since you said it was working before. You might also want to compare your project with the [example repo](https://github.com/vercel/examples/tree/main/solutions/aws-dynamodb) shared above.

If you need more help, please [start a new discussion](https://github.com/vercel/community/discussions/new) with more information about your project and the error message you see.
